prov/efa/src/efa_av.c

```diff
@@ -553,10 +553,38 @@ static int efa_av_remove_ah(struct fid_av *av_fid, fi_addr_t *fi_addr,
 	return ret;
 }
 
+static int efa_dgram_av_remove(struct fid_av *av_fid, fi_addr_t *fi_addr,
+			       size_t count, uint64_t flags)
+{
+	int ret = 0;
+	size_t i;
+	struct efa_av *av;
+
+	av = container_of(av_fid, struct efa_av, util_av.av_fid);
+	for (i = 0; i < count; i++) {
+		ret = efa_av_remove_ah(&av->util_av.av_fid, &fi_addr[i],
+				       1, flags);
+		if (ret)
+			return ret;
+	}
+
+	if (i < count) {
+		/* something went wrong, so ret cannot be zero */
+		assert(ret);
```

Comment:
How can ret be non-zero here if it's already returned at L568?

Reply:
Fixed in new version